src/software/thunderscope/thunderscope_main.py

```diff
@@ -447,6 +446,23 @@ def __ticker(tick_rate_ms: int) -> None:
                 daemon=True,
             )
 
-            thread.start()
-            tscope.show()
-            thread.join()
+            if args.enable_autoref and args.ci_mode:
+                exiter_thread = threading.Thread(
+                    target=exit_poller,
+                    args=(autoref, CI_DURATION_S, lambda: tscope.close()),
+                    daemon=True,
+                )
+
+                exiter_thread.start()
+                thread.start()
```

Comment:
Could you add a comment explaining what's going on here? Also, could we rename `thread` variable to `sim_ticker_thread`?

Reply:
done